compiler-core/src/language_server/engine.rs

```diff
@@ -417,6 +415,44 @@ where
 
         completions
     }
+
+    pub fn inlay_hint(&mut self, params: lsp::InlayHintParams) -> Response<Option<Vec<InlayHint>>> {
+        self.respond(|this| {
+            let Some(module) = this.module_for_uri(&params.text_document.uri) else {
+                return Ok(None);
+            };
+            let line_numbers = LineNumbers::new(&module.code);
+
+            let mut hints = vec![];
+            let start =
+                line_numbers.byte_index(params.range.start.line, params.range.start.character);
+            let end = line_numbers.byte_index(params.range.end.line, params.range.end.character);
+            add_hints_for_definitions(
+                &this.user_config.inlay_hints,
+                module.ast.definitions.iter().filter(|stmt| {
+                    let loc = stmt.location();
+                    (loc.start > start && loc.start < end) // Is the start of the node in the range
+                        || (loc.end > start && loc.end < end) // or is the end of the node in the range
+                        || (start > loc.start && end < loc.end) // or is range within the node
```

Comment:
Could we extract this to a `fn overlaps(self, other: SrcSpan) -> bool` or such please?